src/lib/converter/symbols.ts

```diff
@@ -899,10 +914,24 @@ function convertVariableAsEnum(
             prop,
             declaration
         );
-        assert(propType.isStringLiteral() || propType.isNumberLiteral());
 
-        reflection.defaultValue = JSON.stringify(propType.value);
-        reflection.type = new LiteralType(propType.value);
+        if (propType.isStringLiteral() || propType.isNumberLiteral()) {
+            // The trivial case is a string or number literal.
+            const enumMemberValue = propType.value;
+            reflection.defaultValue = JSON.stringify(enumMemberValue);
+            reflection.type = new LiteralType(enumMemberValue);
+        } else {
+            // If this is not a string or number literal, this it is probably an expression, like
+            // "1 << 1". We cannot safely evaluate the code to find out the value, so revert to
+            // printing "number" or "string", similar to what TypeScript itself does.
+            if (isNumberLike(propType)) {
+                reflection.type = new IntrinsicType("number");
+            } else if (isStringLike(propType)) {
+                reflection.type = new IntrinsicType("string");
+            } else {
+                reflection.type = new IntrinsicType("unknown");
+            }
+        }
```

Comment:
With the extra cases, I think it's worth calling out to `convertType` now

```suggestion
        reflection.type = context.converter.convertType(context, propType);

        if (propType.isStringLiteral() || propType.isNumberLiteral()) {
            reflection.defaultValue = JSON.stringify(propType.value);
        }
```